Python/bytecodes.c

```diff
@@ -143,17 +143,24 @@ dummy_func(
 
         tier1 inst(RESUME, (--)) {
             assert(frame == tstate->current_frame);
-            uintptr_t global_version =
-                _Py_atomic_load_uintptr_relaxed(&tstate->eval_breaker) &
-                ~_PY_EVAL_EVENTS_MASK;
-            uintptr_t code_version = _PyFrame_GetCode(frame)->_co_instrumentation_version;
-            assert((code_version & 255) == 0);
-            if (code_version != global_version) {
-                int err = _Py_Instrument(_PyFrame_GetCode(frame), tstate->interp);
-                ERROR_IF(err, error);
-                next_instr = this_instr;
-            }
-            else {
+            if (tstate->tracing == 0) {
```

Comment:
Ah I remembered why I did not set this instr to `RESUME_CHECK` when `tracing != 0` - it will convert this instr to `RESUME_CHECK` but `RESUME_CHECK` will be deopted as the version does not add up.

For your question - the key is to stop the code from being instrumented, removing this check will still result in an instrumented code, and instrumented code is always slower.

However, we should probably check if the code is tracing and do not deopt `RESUME_CHECK` if so.

Reply:
Another way to do this is to `DISPATCH()` in the instrument case, so we don't have to write the "normal" case twice. Is it frown upon to have multiple `DISPATCH()` in one bytecode?

Reply:
Multiple `DISPATCH()`s should be fine.